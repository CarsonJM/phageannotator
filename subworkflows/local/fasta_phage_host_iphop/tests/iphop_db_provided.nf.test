nextflow_workflow {

    name "Test Subworkflow: FASTA_PHAGE_HOST_IPHOP"
    script "../main.nf"
    workflow "FASTA_PHAGE_HOST_IPHOP"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "fasta_phage_host_iphop"
    tag "fasta_phage_host_iphop_db_provided"
    tag "untar"


    test("Parameters: --iphop_db provided") {

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir              = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of([ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true) ] ])
                input[1] = Channel.of([ file(params.iphop_db, checkIfExists:true ) ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
