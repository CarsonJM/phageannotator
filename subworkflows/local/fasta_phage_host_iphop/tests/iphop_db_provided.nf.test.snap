{
    "Parameters: --iphop_db provided": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,e529062e32656f9e9e1737ebd66c4e83"
                    ]
                ],
                "1": [
                    "versions.yml:md5,41c2135a3578a7a52a23fd3243c22e80",
                    "versions.yml:md5,9a3afe922371942954848605293c0700"
                ],
                "host_predictions_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,e529062e32656f9e9e1737ebd66c4e83"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,41c2135a3578a7a52a23fd3243c22e80",
                    "versions.yml:md5,9a3afe922371942954848605293c0700"
                ]
            }
        ],
        "timestamp": "2023-11-16T16:15:22.426800093"
    }
}